internal/controllers/nmc_reconciler.go

```diff
@@ -651,10 +692,86 @@ func newPodManager(client client.Client, workerImage string, scheme *runtime.Sch
 	}
 }
 
-func (p *podManagerImpl) CreateLoaderPod(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleSpec) error {
+func (p *podManagerImpl) CreateLoaderPod(ctx context.Context, nmcObj client.Object, nms *kmmv1beta1.NodeModuleSpec) error {
+	pod, err := p.MakeLoaderPod(ctx, nmcObj, nms)
+	if err != nil {
+		return fmt.Errorf("could not get loader Pod template: %v", err)
+	}
+
+	return p.client.Create(ctx, pod)
+}
+
+func (p *podManagerImpl) CreateUnloaderPod(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleStatus) error {
+	pod, err := p.MakeUnloaderPod(ctx, nmc, nms)
+	if err != nil {
+		return fmt.Errorf("could not create the Pod template: %v", err)
+	}
+
+	return p.client.Create(ctx, pod)
+}
+
+func (p *podManagerImpl) DeletePod(ctx context.Context, pod *v1.Pod) error {
+	logger := ctrl.LoggerFrom(ctx)
+
+	logger.Info("Removing Pod finalizer")
+
+	podPatch := client.MergeFrom(pod.DeepCopy())
+
+	controllerutil.RemoveFinalizer(pod, nodeModulesConfigFinalizer)
+
+	if err := p.client.Patch(ctx, pod, podPatch); err != nil {
+		return fmt.Errorf("could not patch Pod %s/%s: %v", pod.Namespace, pod.Name, err)
+	}
+
+	if pod.DeletionTimestamp == nil {
+		logger.Info("DeletionTimestamp not set; deleting Pod")
+
+		if err := p.client.Delete(ctx, pod); client.IgnoreNotFound(err) != nil {
+			return fmt.Errorf("could not delete Pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+	} else {
+		logger.Info("DeletionTimestamp set; not deleting Pod")
+	}
+
+	return nil
+}
+
+func (p *podManagerImpl) GetWorkerPod(ctx context.Context, podName, namespace string) (*v1.Pod, error) {
+	pod := v1.Pod{}
+	nsn := types.NamespacedName{Namespace: namespace, Name: podName}
+
+	if err := p.client.Get(ctx, nsn, &pod); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		} else {
+			return nil, fmt.Errorf("could not get pod %s: %v", nsn, err)
+		}
+	}
+
+	return &pod, nil
+}
+
+func (p *podManagerImpl) ListWorkerPodsOnNode(ctx context.Context, nodeName string) ([]v1.Pod, error) {
+	logger := ctrl.LoggerFrom(ctx).WithValues("node name", nodeName)
+
+	pl := v1.PodList{}
+
+	hl := client.HasLabels{actionLabelKey}
+	mf := client.MatchingFields{".spec.nodeName": nodeName}
+
+	logger.V(1).Info("Listing worker Pods")
+
+	if err := p.client.List(ctx, &pl, hl, mf); err != nil {
+		return nil, fmt.Errorf("could not list worker pods for node %s: %v", nodeName, err)
+	}
+
+	return pl.Items, nil
+}
+
+func (p *podManagerImpl) MakeLoaderPod(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleSpec) (*v1.Pod, error) {
```

Comment:
Maybe rename it to LoaderPodTemplate, or something like that? It is a little bit confusing, the expectation from the name is that the pod is actually created

Reply:
WDYT about `LoaderPod`?

Reply:
Done, PTAL.